packages/langium/src/parser/token-builder.ts

```diff
@@ -0,0 +1,94 @@
+/******************************************************************************
+ * Copyright 2021 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import { Lexer, TokenPattern, TokenType } from 'chevrotain';
+import { Grammar, isKeyword, isTerminalRule, TerminalRule } from '../grammar/generated/ast';
+import { streamAllContents } from '../utils/ast-util';
+import { partialMatches } from '../utils/regex-util';
+import { stream } from '../utils/stream';
+
+export interface TokenBuilder {
+    buildTokens(grammar: Grammar): TokenType[];
+}
+
+export class DefaultTokenBuilder implements TokenBuilder {
+
+    buildTokens(grammar: Grammar): TokenType[] {
```

Comment:
Could we keep a generated `TokenType[]` array to speed up the startup time of Langium?

Reply:
Yes/no. This is a premature optimization. I'll keep this in mind if we notice any startup performance issues 👍